packages/ripple-binary-codec/src/enums/README.md

```diff
@@ -31,30 +33,70 @@ Sort code numbers are reused for fields of different types, but different fields
 
 - https://github.com/ripple/ripple-binary-codec/blob/14e76e68ead7e4bcd83c942dbdc9064d5a66869b/src/enums/definitions.json#L832
 - https://github.com/ripple/rippled/search?utf8=%E2%9C%93&q=taker_gets_funded&type=
 
-### isSigningField
+#### isSigningField
 
 True unless the field is [specified with `SField::notSigning`](https://github.com/ripple/rippled/blob/eaff9a0e6aec0ad077f118501791c7684debcfd5/src/ripple/protocol/impl/SField.cpp#L198).
 
-## Transaction Results
+### Transaction Results
 
 See:
 
 - https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/TER.h
 - https://xrpl.org/transaction-results.html
 
-TODO: Write a script to read rippled's source file and generate the necessary mapping.
+To generate a new definitions file from rippled source code, use this tool: https://github.com/RichardAH/xrpl-codec-gen
 
-## Transaction Types
+### Transaction Types
 
 See https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/TxFormats.h
+
+# Defining Your Own Definitions
+
+If you're building your own side chain or amendment for the XRPL, you may need to create new definitions.
+
+To do that there are three high level things you need to do:
+1. Copy and modify the `definitions.json` file to have your definitions ([link](https://github.com/XRPLF/xrpl.js/blob/main/packages/ripple-binary-codec/src/enums/definitions.json)) as a baseline, or you can generate your own from rippled source code with [this tool](https://github.com/RichardAH/xrpl-codec-gen)
+2. Create new SerializedType classes for any new Types (So that encode/decode behavior is defined)
+3. Import `DEFINITIONS` and update it with your new `definitions.json` and SerializedType definitions.
+
+To see this in action, look at the below snippet (Or the test file which contains examples of adding each type of definition)
+
+```
+// DEFINITIONS is the global tracker of type definitions, so we have to update it to add new types
+const { DefinitionContents, DEFINITIONS } = require('../dist/coretypes')
+
+// coreTypes is the default list of serialized Types that are defined in xrpl.js
+const { coreTypes } = require('../dist/types')
+
+// newTypeDefs is where you can import your custom defined definitions.json file
+const newTypeDefs = require('./fixtures/new-type.json')
+
+
+// For any new Types you create, you'll need to make a class with the same name which extends a SerializedType object
+// In order to define how to serialize/deserialize that field. Here we simply make our NewType act like a UInt32.
+
+const { UInt32 } = require('../dist/types/uint-32')
+class NewType extends UInt32 {
+  // Should be the same as UInt32
+}
+
+const extendedCoreTypes = { ...coreTypes }
+extendedCoreTypes['NewType'] = NewType
+
+const newDefs = new DefinitionContents(newTypeDefs)
+DEFINITIONS.updateAll(newDefs, extendedCoreTypes)
+
+// From this point on, we should be able to serialize / deserialize Transactions with fields that have 'NewType' as their Type.
```

Comment:
Does this new API work if you have ripple-binary-codec installed, instead of having the source code locally?

I'm not a huge fan of this API - it requires you to dig into the depths of ripple-binary-codec, which is pretty complicated. Is it possible to instead just accept extra params into the top level methods, like `encode(blob, newDefs, extendedCoreTypes)`?

Reply:
You could then call those same methods in `encode`, instead of asking the user to do so.